src/components/Api.js

```diff
@@ -0,0 +1,133 @@
+export default class Api{
+    constructor(options, loadUserInfoCallback, loadCardsCallback, editProfileCallback, addNewCardCallback){
+      this._baseUrl = options.baseUrl;
+      this._headers = options.headers;
+      this._loadUserInfoCallback = loadUserInfoCallback;
+      this._loadCardsCallback = loadCardsCallback;
+      this._editProfileCallback = editProfileCallback;
+      this._addNewCardCallback = addNewCardCallback;
+    }
+
+    _checkResponse(res){
+      if(res.ok){
+          return res.json();
+        } else{
+          return Promise.reject(`Error: ${res.status}`);
+        }
+    }
+
+    loadUserInfo(){
+      return fetch(`${this._baseUrl}/users/me`, {
+        headers: this._headers
+      })
+      .then(this._checkResponse)
+      .then(data => {
+        this._loadUserInfoCallback(data);
+      })
+    }
+
+    loadCards(){
+      return fetch(`${this._baseUrl}/cards`, {
+        headers: this._headers
+      })
+      .then(this._checkResponse)
+      .then(data => {
+        this._loadCardsCallback(data);
+      })
+    }
+
+    editProfile(inputValues, popup, popupForm){
+      return fetch(`${this._baseUrl}/users/me`, {
+        method: 'PATCH',
+        headers: this._headers,
+        body: JSON.stringify(inputValues)
+      })
+      .then(this._checkResponse)
+      .then(values => {
+        this._editProfileCallback(values);
+        popupForm.close();
+      })
+      .finally(() => {
+        this.renderLoading(false, popup);
```

Comment:
`renderLoading` should not be in `Api`. It is responsible only for fetching and checking the response


So, you'll have only one `then` with `_checkResponse` and nothing else

```js

  loadCards() {
    return fetch(`${this._baseUrl}/cards`, {
      headers: this._headers,
    }).then(this._checkResponse);
  }

  editProfile(inputValues) {
    return fetch(`${this._baseUrl}/users/me`, {
      method: "PATCH",
      headers: this._headers,
      body: JSON.stringify(inputValues),
    }).then(this._checkResponse);
  }

  addNewCard(inputValues) {
    return fetch(`${this._baseUrl}/cards`, {
      method: "POST",
      headers: this._headers,
      body: JSON.stringify(inputValues),
    }).then(this._checkResponse);
  }
```